frontend/src/metabase-lib/v1/parameters/utils/targets.ts

```diff
@@ -58,13 +60,33 @@ export function getParameterTargetField(
   }
 
   if (isConcreteFieldReference(fieldRef)) {
-    const fieldId = fieldRef[1];
-    const resultMetadata = question.getResultMetadata();
-    const fieldMetadata = resultMetadata.find(field => field.id === fieldId);
-    return (
-      metadata.field(fieldId, fieldMetadata?.table_id) ??
-      metadata.field(fieldId)
+    const query = question.query();
+    const stageIndex = -1;
+    const columns = Lib.visibleColumns(query, stageIndex);
+
+    // query and metadata are not available
```

Comment:
a comment is not very clear. does it mean that metadata is not available _yet_? or is this `if (column.length === 0)` statement means that we don't have metadata available?